diff -rupN orig/CMakeLists.txt patched/CMakeLists.txt
--- orig/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ patched/CMakeLists.txt	2020-03-27 13:09:03.000000000 +0100
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(bzip2)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64")
+
+set(SOURCES blocksort.c
+            huffman.c
+            crctable.c
+            randtable.c
+            compress.c
+            decompress.c
+            bzlib.c)
+
+set(PUBLIC_HEADERS bzlib.h)
+
+add_library(bz2 STATIC ${SOURCES})
+
+set_property(TARGET bz2 PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+install(TARGETS bz2
+    ARCHIVE DESTINATION lib
+)
+
+install(FILES ${PUBLIC_HEADERS} DESTINATION include)
